### ISSUE TITLE
GitHub oauth might not return an email

### ISSUE THREAD

**Issue by MatthewScholefield** [frankie567/fastapi-users#511]
## Describe the bug

When a user without a public email set on GitHub performs OAuth, the email from `GitHubOAuth2.get_id_email` is `None` which causes the following validation error when instantiating the `BaseOAuthAccount` pydantic model in `fastapi_users/router/oauth.py:108`:

```
pydantic.error_wrappers.ValidationError: 1 validation error for BaseOAuthAccount
account_email
  none is not an allowed value (type=type_error.none.not_allowed)
```

## To Reproduce

 - Unset any public GitHub emails in your profile:

![image](https://user-images.githubusercontent.com/5875019/108259691-0666dd80-7127-11eb-9a7b-d52dd79d7d46.png)

 - Register as a new OAuth user with your GitHub account

**Expected behavior:** A new user is created
**Actual behavior:** A validation error

## Discussion

I think the API currently assumes every user has an email and uses this for linking accounts. Solutions I was thinking of include:

 - Autogenerate an "oauth" email with some special token like `abc123@__oauth__:github`
 - Making `email` optional in `UserDB` and `BaseOAuthAccount` and removing the unique email index along with making any necessary adjustments to `get_by_email`

**Comment by frankie567** [comment]
Apparently, this has something to do with the scopes and/or the necessity to make a separate API call: https://stackoverflow.com/questions/35373995/github-user-email-is-null-despite-useremail-scope

So, I don't think we need to do over-complicated things, but just fix the [GitHubOAuth2 client that lives in httpx-oauth](https://github.com/frankie567/httpx-oauth/blob/master/httpx_oauth/clients/github.py).

**Comment by MatthewScholefield** [comment]
What are your plans to deal with multiple email accounts? Perhaps iterate until one match is found and merge with that account, and otherwise use the first email to create the new account?

**Comment by frankie567** [comment]
This should be fixed with latest `httpx-oauth` version (0.3.5).
